src/cursor/cur_file.c

```diff
@@ -469,15 +469,20 @@ __curfile_close(WT_CURSOR *cursor)
 
 	cbt = (WT_CURSOR_BTREE *)cursor;
 	CURSOR_API_CALL_PREPARE_ALLOWED(cursor, session, close, cbt->btree);
-	released = false;
```

Comment:
Does remove this explicit assignment open us up to complaints from code analyzers?  If we want it after all, it should be after the `err:` label.

Reply:
> Does remove this explicit assignment open us up to complaints from code analyzers?

There are two reasons to remove that line: first, it's not necessary, because by definition every called function that has a returned side-effect should immediately set the side-effect to its default at the top of the function. Admittedly, that's coding practice, but I think it's important.

Second, if you remove that line, and the called function doesn't set the side-effect correctly, you'll get a complaint from the static analyzer because it's not necessarily initialized.

So, I'd argue it's cleaner to remove it and leads to better static analysis results.

Reply:
Sounds reasonable.  With the merged changes, the changes to `cur_file.c` LGTM